Search Upgrades

vkoves opened the pull request:
# Description

Made two big search improvements.

## 1. More Search Filters

Added a "New" filter and "All Electric" filter. I think in the future I might want to rework this to "Started Reporting In" or something like that, and add other filters, but this is a good start

<img width="1019" height="671" alt="image" src="https://github.com/user-attachments/assets/c458d255-1066-497f-8484-112c33ac63ad" />

## 2.  **Search Autocomplete!**

Tackled issue #18 with autocomplete on the search page and in the header on other pages, making it a lot easier to find buildings and see issues with addresses, making travelling between buildings more fluid.

| Home | Header |
| --- | --- |
| <img width="1396" height="851" alt="image" src="https://github.com/user-attachments/assets/bfdf8bcd-771f-4029-aed7-a7818e50a727" /> | <img width="1029" height="725" alt="image" src="https://github.com/user-attachments/assets/b089f928-3c5a-4b4f-9e61-d3c6a1e9960c" /> |

# Testing Instructions

Please describe the tests/QA that you did to verify your changes. Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] If I added a large new feature, I added it to the release notes (ReleaseNotes.vue)

## Data Update (if applicable):

- [ ] I have followed the [Data Update Checklist](DATA_UPDATE_CHECKLIST.md) for updating to new year's data

<!-- PR template modified from: https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/ -->

vkoves committed:
Add New & All Electric search filters
The new filter is quite slow though, so we might want to add that as a BE python step

vkoves committed:
Merge branch 'minor-fixes' into search-upgrades

vkoves committed:
Merge branch 'main' into search-upgrades

vkoves committed:
Optimize isNewBuilding check and cache historic data on search page

vkoves committed:
Restyle search filters

vkoves commented:
I'm seeing the out of memory issue again, so I think we need to add `is_new` as a Python property, so we don't need to fetch historic data to determine that and do filtering.

Commit:
proposing an alternative to if statements for filtering in Search.vue

hunter-yeago reviewed (changes requested):
Only thing that has to change is the x-scroll on mobile for the input of the search input (the one with width: 20rem;) - otherwise I've proposed a couple of changes to the code. Could go either way really though!

vkoves committed:
Merge branch 'main' into search-upgrades

vkoves committed:
Fix linting errors

vkoves committed:
Merge branch 'main' into search-upgrades

vkoves committed:
Add search autocomplete
Resolves #18

vkoves committed:
Migrate search index to pre-built CSV
Improves build performance, since the CSV won't change unless building
data changes

vkoves committed:
Regenerate Playwright snapshots

vkoves committed:
Merge branch 'main' into search-upgrades

vkoves committed:
Update search index with not submitted buildings
